Describe RateLimit-Remaining using structured-field-values (#22)
ietf-wg-httpapi/ratelimit-headers: draft-ietf-httpapi-ratelimit-headers.md

The "delay-seconds" rule is defined in Section 10.2.4 of {{SEMANTICS}}.

> This specification uses Structured Headers {{!I-D.ietf-httpbis-header-structure}} to specify syntax.

Reviewer: Taken from https://github.com/httpwg/http-extensions/blame/master/draft-ietf-httpbis-proxy-status.md#L80

Reviewer: > @ioggstream The spec we're working off of isn't called "structured headers" any more. It's called "structured field values": https://httpwg.org/http-extensions/draft-ietf-httpbis-header-structure.html (take note of the title of the document and the language within it, not the I-D's name, which is irrelevant) -- is changing back to calling it "headers" intentional?

My mistake. You can fix the original doc too ;)

> This specification uses Structured Headers {{!I-D.ietf-httpbis-header-structure}} to specify syntax.

Reviewer: ```suggestion
This specification uses Structured Headers {{!SF=I-D.ietf-httpbis-header-structure}} to specify syntax.
```

> This specification uses Structured Headers {{!I-D.ietf-httpbis-header-structure}} to specify syntax.

Reviewer: ```suggestion
This specification uses Structured Headers {{!SF=I-D.ietf-httpbis-header-structure}} to specify syntax.
```

> This specification uses Structured Headers {{!SF=I-D.ietf-httpbis-header-structure}} to specify syntax.

Reviewer: ```suggestion
This specification uses Structured Headers {{!SF=RFC8941}} to specify syntax.
```

## RateLimit-Remaining {#ratelimit-remaining-header}

The `RateLimit-Remaining` response field indicates the remaining `quota-units`

> defined in {{request-quota}} associated to the client.

Author: Not sure how I should deal with this little paragraph. It brings up the `quota-units` ABNF rule, which we're not using any more.

The `RateLimit-Remaining` response field indicates the remaining `quota-units`
defined in {{request-quota}} associated to the client.

The header value is
`RateLimit-Remaining` is a Item Structured Field
{{!I-D.ietf-httpbis-header-structure}}. Its value MUST be an Integer (Section
3.3.1 of {{I-D.ietf-httpbis-header-structure}}). Its ABNF is:

~~~
   RateLimit-Remaining = quota-units
   RateLimit-Remaining = sf-integer
~~~

This header MUST NOT occur multiple times
and can be sent in a trailer section.
The value of `RateLimit-Remaining` MUST be non-negative.

Clients MUST NOT assume that a positive `RateLimit-Remaining` value is
a guarantee of being served.

A low `RateLimit-Remaining` value is like a yellow traffic-light: the red light
may arrive suddenly.
`RateLimit-Remaining` MUST NOT occur multiple times. It MAY be sent in a trailer
section.

> One example of `RateLimit-Remaining` use is below.

Reviewer: While I find your form more readable, other RFCs use this formula "X example[s] of ... is below". This is possibily related to the fact that in case of multiple lines, the reader might not know beforehand whether we are providing one or more examples.

Have a look at https://httpwg.org/http-core/draft-ietf-httpbis-semantics-latest.html#rfc.section.5.6.7.p.3 . Probably we could `s,One,An`

Author: In this spirit of your first suggestion, maybe we can do this in an editorial pass later? That way I can make the diff as small as possible.

> 

Reviewer: @ucarion Thanks for your PR. Just a work suggestion: every changes unrelated to structured headers in this PR distracts the reader from the focal point of this PR.

It is better to address editorial changes (eg. rewording, moving text around, ...) in specific PRs labeled with `editorial`: we can create one after we move all fields to SF.

Does it seem reasonable to you? Thanks again! R.

The `RateLimit-Remaining` response field indicates the remaining `quota-units` defined in {{request-quota}}
associated to the client.

> `RateLimit-Remaining` is a Item Structured Field

Reviewer: ```suggestion
`RateLimit-Remaining` is an Item Structured Field
```

The `RateLimit-Remaining` response field indicates the remaining `quota-units` defined in {{request-quota}}
associated to the client.

The header value is
`RateLimit-Remaining` is an Item {{SF}}.
Its value MUST be a non-negative Integer.

~~~

>    RateLimit-Remaining = quota-units

Reviewer: @reschke @ucarion  as `quota-units` is a recurring element, can we define it in terms of `sf-integer` and then say something like:

```
RateLimit-Remaining is an Item {{SF}}:

~~~ abnf
   RateLimit-Remainig = sf-item
~~~

Its value is defined by `quota-units`...
```

Author: As an alternative approach, could we instead SFV-ify everything and then have "quota units" be a prosaic concept, rather than an ABNF rule?

Reviewer: My mistake, fix it. Should probably fix the referred file under http-extensions
